### PR TITLE
Add a method for calculating num_edges

### PR DESCRIPTION
**Pull request by elham-azd**
Closes https://github.com/dwavesystems/dwave-optimization/issues/28

### CODE REVIEW

File: dwave/optimization/model.pyx [hunk 1]
```diff
@@ -694,6 +694,23 @@ cdef class Model:
...
+            nodes in the model.
+
+            >>> from dwave.optimization.model import Model
+            >>> model = Model()
+            >>> c = model.constant([1, 5, 8.4])
+            >>> i = model.integer(20, upper_bound=100)
+            >>> model.num_edges()
+            0
+        """
+        return sum(1 for sym in self.iter_symbols() for _ in sym.iter_successors())
```
Comment (arcondello) [inline dwave/optimization/model.pyx:712]: We might as well do the more performant thing. Something like
```Cython
        cdef Py_ssize_t num_edges = 0
        for i in range(self._graph.num_nodes()):
            num_edges += self._graph.nodes()[i].successors().size()
        return num_edges
```

File: dwave/optimization/model.pyx [hunk 2]
```diff
@@ -694,6 +694,23 @@ cdef class Model:
...
+            This example add a single (size 20) decision symbol and
+            a single (size 3) constant symbol checks the number of
+            nodes in the model.
+
+            >>> from dwave.optimization.model import Model
+            >>> model = Model()
+            >>> c = model.constant([1, 5, 8.4])
+            >>> i = model.integer(20, upper_bound=100)
+            >>> model.num_edges()
+            0
```
Comment (arcondello) [inline dwave/optimization/model.pyx:710]: Example should probably show at least one edge
```
            >>> from dwave.optimization.model import Model
            >>> model = Model()
            >>> c = model.constant([1, 5, 8.4])
            >>> i = model.integer(3, upper_bound=100)
            >>> model.minimize(c + i)
            >>> model.num_edges()
            2
```

**Review by arcondello — COMMENTED** [review]
Also needs a release note.

File: dwave/optimization/model.pyx [hunk 3]
```diff
@@ -694,6 +694,27 @@ cdef class Model:
         """
         return self._graph.num_decisions()
 
+    def num_edges(self):
+        """Number of edges in the directed acyclic graph for the model.
+
+        Examples:
+            This example add a single (size 20) decision symbol and
+            a single (size 3) constant symbol checks the number of
```
Comment (arcondello) [inline dwave/optimization/model.pyx:702]: I think these numbers are incorrect?

File: releasenotes/notes/add-num_edges-4569f47ca86590de.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Adds a method for calculating num_edges
```
Comment (arcondello) [inline releasenotes/notes/add-num_edges-4569f47ca86590de.yaml:4]: Maybe something like
```yaml
    Add `Model.num_edges()` method. See `#28 <https://github.com/dwavesystems/dwave-optimization/issues/28>`_.
```

File: dwave/optimization/model.pyx (hunk 3 above)
Comment (elham-azd) [reply dwave/optimization/model.pyx]: Ughh, forgot to update this. Thanks!

File: releasenotes/notes/add-num_edges-4569f47ca86590de.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    Add `Model.num_edges()` method.
+    See `#28 <https://github.com/dwavesystems/dwave-optimization/issues/28>`
```
Comment (arcondello) [inline releasenotes/notes/add-num_edges-4569f47ca86590de.yaml:5]:
```suggestion
    See `#28 <https://github.com/dwavesystems/dwave-optimization/issues/28>`_.
```

**Review by arcondello — APPROVED** [review]

**Pull request merged**